plan/physical_plan_builder.go

```diff
@@ -245,7 +245,12 @@ func (p *DataSource) convert2IndexScan(prop *requiredProperty, index *model.Inde
 		if client != nil {
 			memDB := infoschema.IsMemoryDB(p.DBName.L)
 			if !memDB && client.SupportRequestType(kv.ReqTypeIndex, 0) {
-				is.ConditionPBExpr, is.conditions, newSel.Conditions = expressionsToPB(sc, newSel.Conditions, client)
+				var indexConditions, pbConditions []expression.Expression
+				indexConditions, newSel.Conditions = detachIndexFilterConditions(newSel.Conditions, is.Index.Columns, is.Table)
+				is.IndexPBExpr, is.conditions, indexConditions = expressionsToPB(sc, indexConditions, client)
+				is.ConditionPBExpr, pbConditions, newSel.Conditions = expressionsToPB(sc, newSel.Conditions, client)
+				newSel.Conditions = append(newSel.Conditions, indexConditions...)
+				is.conditions = append(is.conditions, pbConditions...)
```

Comment:
You can split is.conditions into two parts: IndexFilterConditions and TableFilterConditions. And show them in explain result.